sql/src/main/java/io/crate/analyze/BoundedCopyFrom.java

```diff
@@ -24,64 +24,59 @@
 
 import io.crate.execution.dsl.phases.FileUriCollectPhase;
 import io.crate.expression.symbol.Symbol;
-import org.elasticsearch.cluster.node.DiscoveryNode;
 import io.crate.metadata.doc.DocTableInfo;
-import io.crate.types.DataType;
+import org.elasticsearch.cluster.node.DiscoveryNode;
 import org.elasticsearch.common.settings.Settings;
 
 import javax.annotation.Nullable;
 import java.util.function.Predicate;
 
-public class CopyFromAnalyzedStatement extends AbstractCopyAnalyzedStatement {
+public class BoundedCopyFrom {
 
-    private final DocTableInfo table;
+    private final DocTableInfo tableInfo;
     @Nullable
     private final String partitionIdent;
-    private final Predicate<DiscoveryNode> nodePredicate;
-    @Nullable
+    private final Settings settings;
+    private final Symbol uri;
     private final FileUriCollectPhase.InputFormat inputFormat;
+    private final Predicate<DiscoveryNode> nodeFilters;
 
-    public CopyFromAnalyzedStatement(DocTableInfo table,
-                                     Settings settings,
-                                     Symbol uri,
-                                     @Nullable String partitionIdent,
-                                     Predicate<DiscoveryNode> nodePredicate,
-                                     FileUriCollectPhase.InputFormat inputFormat) {
-        super(settings, uri);
-        this.table = table;
+    public BoundedCopyFrom(DocTableInfo tableInfo,
+                           @Nullable String partitionIdent,
+                           Settings settings,
+                           Symbol uri,
```

Comment:
still symbol, will follow up later with a PR where we have `uri` as `List<String>`